test/core/util/port_isolated_runtime_environment.cc

```diff
@@ -28,18 +29,21 @@
 #include "test/core/util/port.h"
 
 #define MIN_PORT 49152
-#define MAX_PORT 65536
+#define MAX_PORT 65535
 
 int get_random_starting_port() {
   srand(gpr_now(GPR_CLOCK_REALTIME).tv_nsec);
-  return rand() % (MAX_PORT - MIN_PORT + 1) + MIN_PORT;
+  double rnd = static_cast<double>(rand()) /
+               (static_cast<double>(RAND_MAX) + 1.0);  // values from [0,1)
+  return static_cast<int>(rnd * (MAX_PORT - MIN_PORT + 1)) + MIN_PORT;
 }
 
 static int s_allocated_port = get_random_starting_port();
 
 int grpc_pick_unused_port_or_die(void) {
+  // TODO(jtattermusch): protect by mutex
```

Comment:
@yashykt I wanted to protect by mutex as a safety measure, but wasn't sure how to handle the mutex initialization and destroy without hooking into the test's init and cleanup logic. Any tips?

Reply:
we could use atomics instead